File: Project-Euler/Solutions-Simon/euler3/euler3.jl
#=
  The prime factors of 13195 are 5, 7, 13 and 29.
  What is the largest prime factor of the number 600851475143.
=#
function printsum(a)
    # summary generates a summary of an object
    println(repr(a))
end

function prime_factors(n)
  factors = Int64[]
  d = 2
  while n > 1
    while n % d == 0
      push!(factors, d)
      n /= d
    end
    d += 1
  end
  printsum(factors)
end

prime_factors(600851475143)

File: Project-Euler/Solutions-Simon/euler3/euler3.java
import java.util.*;

public class euler3
{
	public static void main (String[] args)
	{
		System.out.println(prime_factors(600851475143l));
	}
	
	private static List<Long> prime_factors(long n)
	{
		List<Long> factors = new ArrayList<>();
		long d = 2;
		while(n > 1)
		{
			while(n % d == 0)
			{
				factors.add(d);
				n /= d;
			}
			d += 1;
		}
		return factors;
	}
}
